src/main.rs

```diff
@@ -844,7 +844,18 @@ fn construct_backtrace(
             println!("      <exception entry>");
 
             let sp = registers.get(SP)?;
-            let stacked = Stacked::read(registers.core, sp, fpu)?;
+            let ram_bounds = sp_ram_region
+                .as_ref()
+                .map(|ram_region| ram_region.range.clone())
+                // if no device-specific information, use the range specific in the Cortex-M* Technical Reference Manual
+                .unwrap_or(0x2000_0000..0x4000_0000);
```

Comment:
although this is what the TRM specifies I know some vendors use `0x1000_0000` for CCRAM (core-coupled RAM)

Reply:
That seems like somewhere you might want to put the stack... hmm, can we use the initial value of SP here and round it down to a multiple of `0x1000_0000`?

Reply:
I have only ever seen the start of RAM being aligned to `0x1000_0000` but I'm not aware of that being a requirement of the architecture. I do know that VTOR needs to be aligned to a power of 2 that's proportional to the number of vectors in the vector table (learned that one the hard way).

That being said, is there any reason the `sp_ram_region` value can be `None`: that is it's not possible to locate the RAM region the SP belongs to? Perhaps the RAM region data that comes from `probe-rs` is not reliable? Unsure if that info is retrieved from the device itself or hardcoded in `probe-rs`.

Reply:
I believe this data comes from a fixed database in probe-rs, which in turn is derived from vendor-provided documentation, so it might not be correct. I haven't seen that happen so far though.

If we do want to be smarter with the fallback here, this isn't really something that needs fixing before this can be merged though.